src/main/java/com/medallia/word2vec/Word2VecModel.java

```diff
@@ -17,29 +17,32 @@
 import com.google.common.primitives.Doubles;
 import com.medallia.word2vec.thrift.Word2VecModelThrift;
 import com.medallia.word2vec.util.Common;
+import org.apache.log4j.Logger;
 
 /**
  * Represents the Word2Vec model, containing vectors for each word
- * <p>
+ * <p/>
  * Instances of this class are obtained via:
  * <ul>
  * <li> {@link #trainer()}
  * <li> {@link #fromThrift(Word2VecModelThrift)}
  * </ul>
- * 
+ *
  * @see {@link #forSearch()}
  */
 public class Word2VecModel {
+	static Logger logger = Logger.getLogger(Word2VecModel.class);
```

Comment:
We have a ProfilingTimer abstraction which lets the user decide if he wants that info logged or e.g. displayed on a UI.  See notes below ...